continuous_integration/recipes/distributed/meta.yaml

```
@@ -3,79 +3,103 @@
-    - tblib >=1.6.0
-    - toolz >=0.8.2
-    - tornado >=5  # [py<38]
-    - tornado >=6.0.3  # [py>=38]
-    - zict >=0.1.3
-    - setuptools <60.0.0
-
-  run_constrained:
-    - openssl !=1.1.1e
 
-test:
-  imports:
-    - distributed
-    - distributed.cli
-    - distributed.comm
-    - distributed.deploy
-    - distributed.diagnostics
-    - distributed.protocol
-  commands:
-    - pip check
-    - dask-scheduler --help
-    - dask-ssh --help
-    - dask-worker --help
-  requires:
-    - pip
+outputs:
+  - name: distributed-proc
+    version: {{ proc_version }}
+    build:
+      number: {{ proc_build_number }}
+      string: {{ build_ext }}
+    test:
+      commands:
+        - exit 0
+    about:
+      home: https://distributed.dask.org
+      summary: A meta-package to select Distributed pre-release build variant
+      license: BSD-3-Clause
+      license_family: BSD
+      license_file: LICENSE.txt
 
-about:
-  home: https://distributed.dask.org
-  summary: Distributed scheduler for Dask
-  license: BSD-3-Clause
-  license_family: BSD
-  license_file: LICENSE.txt
-  description: |
-    Distributed is a lightweight library for distributed computing in Python.
-    It extends both the concurrent.futures and dask APIs to moderate sized
-    clusters.
-  doc_url: https://distributed.dask.org
-  doc_source_url: https://github.com/dask/distributed/blob/main/docs/source/index.rst
-  dev_url: https://github.com/dask/distributed
+  - name: distributed
+    version: {{ version }}
+    build:
+      number: {{ GIT_DESCRIBE_NUMBER }}
+      string: py{{ python | replace(".", "") }}_{{ GIT_DESCRIBE_HASH }}_{{ GIT_DESCRIBE_NUMBER }}_{{ build_ext }}
+      script: python -m pip install . -vv                                           # [not cython_enabled]
+      script: python -m pip install . -vv --install-option="--with-cython=profile"  # [cython_enabled]
+      track_features: {{ "[cythonized-scheduler]" if cython_enabled else "" }}
+      entry_points:
+        - dask-scheduler = distributed.cli.dask_scheduler:go
+        - dask-ssh = distributed.cli.dask_ssh:go
+        - dask-worker = distributed.cli.dask_worker:go
+    requirements:
+      build:
+        - python                                 # [build_platform != target_platform]
+        - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+        - compilers                              # [cython_enabled]
+        - cython >=0.29,<0.30                    # [cython_enabled]
+      host:
+        - python
+        - pip
+      run:
+        - python
+        - click >=6.6
+        - cloudpickle >=1.5.0
+        - cytoolz >=0.8.2
+        - dask-core >={{ dask_version }}
+        - jinja2
+        - msgpack-python >=0.6.0
+        - packaging >=20.0
+        - psutil >=5.0
+        - pyyaml
+        - sortedcontainers !=2.0.0,!=2.0.1
+        - tblib >=1.6.0
+        - toolz >=0.8.2
+        - tornado >=5  # [py<38]
+        - tornado >=6.0.3  # [py>=38]
```

As we just dropped Python 3.7, think we can simplify this to

```suggestion
        - tornado >=6.0.3
```

---

Anything else we should do to follow up on this so that we're only building for a single Python version?

---

We might want to add `noarch: python  # [nocython]` so that we get `noarch` packages for pure Python. Then we can drop the conversion step as well.